pkg/scraper/client/resource/client.go

```diff
@@ -112,12 +111,12 @@ func (kc *kubeletClient) getMetrics(ctx context.Context, url, nodeName string) (
 	buf.Reset()
 	_, err = io.Copy(buf, response.Body)
 	if err != nil {
-		kc.buffers.Put(b)
```

Comment:
staticcheck:
> [SA6002](https://staticcheck.io/docs/checks#SA6002): argument should be pointer-like to avoid allocations (staticcheck)

Reply:
:warning: I'm not totally sure of my fix.

Reply:
Don't agree with the check statement, allocations are not always bad. And don't recommend changing this code without allocation benchmarking.